taxonomy/signals/handlers.py

```diff
@@ -5,10 +5,18 @@
 import logging
 
 from django.dispatch import receiver
+from taxonomy.choices import ProductTypes
 
 from taxonomy.tasks import update_course_skills, update_program_skills, update_xblock_skills
+from taxonomy.utils import delete_product, duplicate_xblock_skills
```

Comment:
I think these might need to be wrapped into celery tasks like the `update_*_skills` tasks above and be called using the `.delay()` function to avoid blocking the main service thread and have the processing happen in the background.